### ISSUE TITLE
Add flow analysis support for `??` and `??=`

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#38679]
For example, in the following code, `z` is not definitely assigned, so the reference to `z` in `print(z.isEven)` should be flagged as an error:
```dart
void test(int? x, int? y) {
  int z;
  x ?? (z = 0);
  x ??= (z = 1);
  print(z.isEven);
}
```

**Comment by stereotype441** [comment]
On further inspection, it looks like we have support for `??`, but not for `??=`.

**Comment by stereotype441** [comment]
Fixed in 8f5a45c22d7282c66b2a38701a23d1753fca78a2.
